src/AuthLDAP.php

```diff
@@ -3010,22 +3026,48 @@ public static function connectToServer(
 
         $ds = @ldap_connect($host, intval($port));
         if ($ds) {
-            @ldap_set_option($ds, LDAP_OPT_PROTOCOL_VERSION, 3);
-            @ldap_set_option($ds, LDAP_OPT_REFERRALS, 0);
-            @ldap_set_option($ds, LDAP_OPT_DEREF, $deref_options);
-            @ldap_set_option($ds, LDAP_OPT_NETWORK_TIMEOUT, $timeout);
-
+            $ldap_options = [
+                LDAP_OPT_PROTOCOL_VERSION => 3,
+                LDAP_OPT_REFERRALS        => 0,
+                LDAP_OPT_DEREF            => $deref_options,
+                LDAP_OPT_NETWORK_TIMEOUT  => $timeout
+            ];
             if (!empty($tls_certfile) && file_exists($tls_certfile)) {
-                @ldap_set_option(null, LDAP_OPT_X_TLS_CERTFILE, $tls_certfile);
+                $ldap_options[LDAP_OPT_X_TLS_CERTFILE] = $tls_certfile;
             }
-
             if (!empty($tls_keyfile) && file_exists($tls_keyfile)) {
-                @ldap_set_option(null, LDAP_OPT_X_TLS_KEYFILE, $tls_keyfile);
+                $ldap_options[LDAP_OPT_X_TLS_KEYFILE] = $tls_keyfile;
+            }
+
+            foreach ($ldap_options as $option => $value) {
+                if (!@ldap_set_option($ds, $option, $value)) {
+                    trigger_error(
+                        static::buildError(
+                            $ds,
+                            sprintf(
+                                "Unable to set LDAP option %s to %s",
+                                $option,
+                                $value
+                            )
+                        ),
+                        E_USER_WARNING
+                    );
+                }
             }
 
             if ($use_tls) {
                 if (!@ldap_start_tls($ds)) {
-                    return false;
+                    trigger_error(
+                        static::buildError(
+                            $ds,
+                            sprintf(
+                                "Unable to start TLS connection to LDAP server %s:%s",
+                                $host,
+                                $port
+                            )
+                        ),
+                        E_USER_WARNING
+                    );
```

Comment:
We should probably stop here.
```suggestion
                    );
                    return false;
```